### PR TITLE
fix(nfl): derive home_opening_kickoff in enrich for the 4th-down decision step

### PR DESCRIPTION
**Pull request by saiemgilani**
## Problem

`enrich_nfl_pbp`'s 4th-down decision step needs `home_opening_kickoff` (it sets `home_receive_2h_ko` for the nfl4th features). nflverse pbp and the ESPN construction path carry it, but **native producers reconstructed from the Shield feed (`native_pbp`) do not** — so `get_4th_down_probs` raised `KeyError: 'home_opening_kickoff'`, the whole decision step was skipped (with a `RuntimeWarning`), and `go_wp` / `go_boost` / `fourth_down_recommendation` etc. shipped all-null in the published `nfl_model_pbp`.

## Fix

Derive `home_opening_kickoff` (= 1 iff the home team had the game's first possession) before the 4th-down step, when absent. Mirrors the existing `receive_2h_ko` derivation in `_add_wp_aux`:
- derived only when absent — never overrides an nflverse/ESPN-provided column;
- the per-game first-possession lookup (`posteam.drop_nulls().first().over("game_id")`) runs on the full frame before the 4th-down subset filter;
- skips the opening-kickoff row's null `posteam` (first NON-null possession wins).

## Validation

- **End-to-end** on a real `native_pbp` 2024 frame with no `home_opening_kickoff`: no skip warning, `home_opening_kickoff` derived, all 22 4th-down plays get a `fourth_down_recommendation` (`field_goal`/`punt`/`go`).
- 3 derivation unit tests (`test_enrich_derive.py`) + 128 enrich / 4th-down / ep_wp regression tests pass; mypy + ruff clean.

## Context

Surfaced by the sdv-py data-validation harness while reconciling the NFL producer (companion to the `yardline_100` and `spread_line`/`total_line` producer fixes in sportsdataverse/nfl-data). This is the sdv-py-side piece so the 4th-down surface populates for native producers once shipped.

## Summary by Sourcery

Ensure NFL fourth-down decision enrichment derives home_opening_kickoff when missing so 4th-down recommendation features populate for native play-by-play producers.

New Features:
- Add helper to derive home_opening_kickoff from first possession data when not supplied by upstream producers.

Tests:
- Add unit tests covering home_opening_kickoff derivation, handling of leading null posteam values, and preservation of existing upstream values.

### CODE REVIEW

File: sportsdataverse/nfl/ep_wp.py [hunk 1]
```diff
@@ -2867,6 +2867,28 @@ def _flag(col: str) -> pl.Expr:
...
+    The nfl4th 4th-down decision features require it (it sets ``home_receive_2h_ko``).
+    nflverse pbp and the ESPN construction path carry it, but native producers
+    reconstructed from the Shield feed (``native_pbp``) do not — without it
+    :func:`_add_fourth_down_decisions` skips the whole decision step on a
+    ``KeyError``.  Mirrors the ``receive_2h_ko`` derivation in :func:`_add_wp_aux`:
+    derived only when absent (never overrides an upstream column), and only when
+    the source columns exist.
+    """
+    if "home_opening_kickoff" in df.columns or "posteam" not in df.columns or "home_team" not in df.columns:
+        return df
```
[the remark this answers is not in the document]
Comment (saiemgilani) [reply sportsdataverse/nfl/ep_wp.py:2890]: Addressed in 0ce553bf: added `game_id`/`play_id` to the early-return guard and made the first-possession lookup deterministic via `posteam.sort_by("play_id").drop_nulls().first().over("game_id")` (note: `sort_by` must precede `drop_nulls` — the reversed order mismatches lengths). Also cast to Float64 to match nflverse's `home_opening_kickoff` dtype. Thanks!

File: sportsdataverse/nfl/ep_wp.py [hunk 2]
```diff
@@ -2867,6 +2867,28 @@ def _flag(col: str) -> pl.Expr:
...
+    return df.with_columns(
+        pl.when(pl.col("posteam").drop_nulls().first().over("game_id") == pl.col("home_team"))
+        .then(1)
+        .otherwise(0)
+        .alias("home_opening_kickoff"),
+    )
```
[the remark this answers is not in the document]
Comment (saiemgilani) [reply sportsdataverse/nfl/ep_wp.py]: Addressed in 0ce553bf: added `game_id`/`play_id` to the early-return guard and made the first-possession lookup deterministic via `posteam.sort_by("play_id").drop_nulls().first().over("game_id")` (note: `sort_by` must precede `drop_nulls` — the reversed order mismatches lengths). Also cast to Float64 to match nflverse's `home_opening_kickoff` dtype. Thanks!
